src/pages/dataElements/edit/createJsonPatchOperations.spec.ts

```diff
@@ -0,0 +1,127 @@
+                'attributeValues[1].value',
+            ])
+            const expected = ['foo', 'bar', 'attributeValues']
+            expect(actual).toEqual(expected)
+        })
+
+        it('should remove style.icon and style.color changes and add a single "style"', () => {
+            const actual = sanitizeDirtyValueKeys([
+                'foo',
+                'bar',
+                'style.color',
+                'style.icon',
+            ])
+            const expected = ['foo', 'bar', 'style']
+            expect(actual).toEqual(expected)
+        })
+    })
+
+    describe('createJsonPatchOperations', () => {
+        it('should return an empty array if no dirty fields', () => {
+            const actual = createJsonPatchOperations({
+                dirtyFields: {},
+                originalValue: {
+                    id: 'foo',
+                    attributeValues: [],
+                },
+                values: {
+                    attributeValues: [],
+                },
+            })
+            expect(actual).toEqual([])
+        })
+
+        it('should return a json-patch payload for a single field', () => {
+            const actual = createJsonPatchOperations({
+                dirtyFields: {
+                    name: true,
+                },
+                originalValue: {
+                    id: 'foo',
+                    name: 'bar',
+                    attributeValues: [],
+                },
+                values: {
+                    name: 'baz',
+                    attributeValues: [],
+                },
+            })
+            const expected = [
+                {
+                    op: 'replace',
+                    path: '/name',
+                    value: 'baz',
+                },
+            ]
+            expect(actual).toEqual(expected)
+        })
+        it('should return a json-patch payload with add if value does not exist in originalValue', () => {
+            const actual = createJsonPatchOperations({
+                dirtyFields: {
+                    name: true,
+                },
+                originalValue: {
+                    id: 'foo',
+                    attributeValues: [],
+                },
+                values: {
+                    name: 'baz',
+                    attributeValues: [],
+                },
+            })
+            const expected = [
+                {
+                    op: 'add',
+                    path: '/name',
+                    value: 'baz',
+                },
+            ]
+            expect(actual).toEqual(expected)
+        })
+
+        it('should handle attributeValues', () => {
+            const actual = createJsonPatchOperations({
+                dirtyFields: {
+                    attributeValues: true,
+                },
+                originalValue: {
+                    id: 'foo',
+                    name: 'bar',
+                    attributeValues: [],
```

Comment:
Technically this is correct. In reality this will never be an empty array but prefilled with all existing attributes with empty value (see [here](https://github.com/dhis2/maintenance-app-beta/blob/fix/editing-style-object/src/pages/dataElements/New.tsx#L28:L35) and [here](https://github.com/dhis2/maintenance-app-beta/blob/fix/editing-style-object/src/pages/dataElements/Edit.tsx#L65:L76)). I think by adding the attribute with empty value this test also serves as documentation implicitly.

```suggestion
                    attributeValues: [
                        { value: '', attribute: { id: 'foo' } },
                    ],
```